spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/AbstractSpannerRequestOptions.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.core;
+
+import java.io.Serializable;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import com.google.cloud.Timestamp;
+import com.google.cloud.spanner.TimestampBound;
+
+/**
+ * Abstract class of common Read and Query request settings.
+ *
+ * @author Chengyuan Zhao
+ */
+public class AbstractSpannerRequestOptions<A> implements Serializable {
+
+	protected transient List<A> requestOptions = new ArrayList<>();
+
+	protected Class<A> requestOptionType;
```

Comment:
Does this actually work? This member is never assigned? Would it not be `null`?

Reply:
It's assigned in the constructors of the child classes. It's needed to create that array of ReadOption or QueryOption that the client lib requires.